src/Sarif.UnitTests/Processors/Log/RebaseUriStageTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Processors
+{
+    public class RebaseUriStageTests
+    {
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(5)]
+        public void RewriteUri_RewritesAllFiles(int fileCount)
+        {
+            // Slightly roundabout.  We want to randomly test this, but we also want to be able to repeat this if the test fails.
+            int randomSeed = (new Random()).Next();
+            Random random = new Random(randomSeed);
```

Comment:
put all this construction in a helper, then you can add logging there.

Reply:
or maybe just move this into the RandomSarifLogGenerator, i.e., have the Generate method just create the Random objects and log the seed value. 

this suggestion might not be a good one, depending on whether it would require plumbing through some output/logging object.

---
In reply to: [175206341](https://github.com/Microsoft/sarif-sdk/pull/803#discussion_r175206341) [](ancestors = 175206341)

Reply:
Done.